app/controllers/collections_controller.rb

```diff
@@ -3,11 +3,10 @@
 require 'zip'
 
 class CollectionsController < ApplicationController
-  before_action :load_and_authorize_collection, except: %i[index new create]
+  before_action :load_and_authorize_collection, except: %i[index new create create_ajax]
 
   def index
-    @collections = Collection.includes(:users, tasks: %i[user groups])
-      .where(collection_users: {user: current_user})
+    @collections = Collection.member(current_user).or(Collection.public_access).includes(:users, tasks: %i[user groups])
```

Comment:
It's hard to tell whether this is related to the AJAX action or not. Usually, I try to keep the changes separate, but probably not needed here.

Reply:
So this will be extracted or are you planning to keep it here, now?